packages/web-components/fast-components/src/select/index.ts

```diff
@@ -1,9 +1,92 @@
 import {
-    Select,
+    Select as FoundationSelect,
     SelectOptions,
     selectTemplate as template,
 } from "@microsoft/fast-foundation";
-import { selectStyles as styles } from "./select.styles";
+import { optionHeight } from "../listbox-option/listbox-option.styles";
+import { listboxSize } from "../listbox/listbox.styles";
+import {
+    selectMaxSize,
+    selectOptionWidth,
+    selectStyles as styles,
+} from "./select.styles";
+
+/**
+ * Base class for Select.
+ * @public
+ */
+export class Select extends FoundationSelect {
+    /**
+     * Sets the listboxSize design token when the multiple property changes.
+     *
+     * @param prev - the previous multiple value
+     * @param next - the current multiple value
+     *
+     * @override
+     * @internal
+     */
+    public multipleChanged(prev: boolean | undefined, next: boolean): void {
+        super.multipleChanged(prev, next);
+        listboxSize.setValueFor(this, this.size || (next ? 4 : 0));
+    }
+
+    /**
+     * Sets the selectMaxSize design token when the maxHeight property changes.
```

Comment:
Would it help if this were just plain css and not based on the exact height of items? If it's a long list and it's going to scroll it matters less that the items exactly fit than what else it probably aligns with, i.e., a full list on a left side pane or something. Seeing the top half of the item is fine and encourages scrolling.

Reply:
Light-DOM styles should still be able to override any values configured by the `size` attribute. The WHATWG HTML spec describes the [display size](https://html.spec.whatwg.org/multipage/form-elements.html#concept-select-size) but doesn't specify how it should be visually expressed, so I made it match the behavior of the native `<select>` within reason.

I updated the PR to reduce the amount of design tokens, and created one called `heightNumberAsToken` which is only used to get our expected control height (and isn't available as a CSS property, so the `heightNumber` partial is still the main way to set a control's height). If any options deviate from that value, authors can override the height of the control with CSS.